Applications/Spire/Source/Ui/QuantityBox.cpp

```diff
@@ -14,14 +16,65 @@ namespace {
   };
 }
 
+struct QuantityBox::UnsignedQuantityModel : OptionalQuantityModel {
+  mutable CurrentSignal m_current_signal;
+  std::shared_ptr<OptionalQuantityModel> m_model;
+  scoped_connection m_current_connection;
+
+  UnsignedQuantityModel(std::shared_ptr<OptionalQuantityModel> model)
+    : m_model(std::move(model)),
+      m_current_connection(m_model->connect_current_signal(
+        [=] (const auto& current) {
+          on_current(current);
+        })) {}
+
+  optional<Quantity> get_minimum() const override {
+    return 0;
```

Comment:
This is too permissive of a condition. When implementing models that add constraints, you must respect the constraints of the original model in addition to any new constraints.